packages/aws-cdk-lib/aws-logs/lib/log-retention-provider/index.ts

```diff
@@ -206,19 +174,41 @@ export async function handler(event: AWSLambda.CloudFormationCustomResourceEvent
       }
     });
   }
+}
 
-  function parseRetryOptions(rawOptions: any): SdkRetryOptions {
-    const retryOptions: SdkRetryOptions = {};
-    if (rawOptions) {
-      if (rawOptions.maxRetries) {
-        retryOptions.maxRetries = parseInt(rawOptions.maxRetries, 10);
-      }
-      if (rawOptions.base) {
-        retryOptions.retryOptions = {
-          base: parseInt(rawOptions.base, 10),
-        };
-      }
-    }
-    return retryOptions;
+function parseIntOptional(value?: string, base = 10): number | undefined {
+  if (value === undefined) {
+    return undefined;
   }
+
+  return parseInt(value, base);
+}
+
+function makeWithDelay(maxRetries: number = 10, delay: number = 100): (block: () => Promise<void>) => Promise<void> {
+  // If we try to update the log group, then due to the async nature of
+  // Lambda logging there could be a race condition when the same log group is
+  // already being created by the lambda execution. This can sometime result in
+  // an error "OperationAbortedException: A conflicting operation is currently
+  // in progress...Please try again."
+  // To avoid an error, we do as requested and try again.
+
+  return async (block: () => Promise<void>) => {
+    do {
+      try {
+        return await block();
+      } catch (error: any) {
+        if (error instanceof Logs.OperationAbortedException || error.name === 'OperationAbortedException') {
+          if (maxRetries > 0) {
+            maxRetries--;
+            await new Promise(resolve => setTimeout(resolve, delay));
```

Comment:
Can we  make this exponential delay?

Reply:
```
delay *= 2;
```

Reply:
Going with 10s as a max now. This creates this distribution for the default of 10 rounds, for a total of ~20s.
```
[
    60,    70,   90,
   130,   210,  370,
   690,  1330, 2610,
  5170, 10000
]
20730ms
```